### PR TITLE
Blog App: validations and model specs

### PR DESCRIPTION
**Pull request by akos29**
Blog App: validations and model specs
--

In this request majorly:

- Add validations for the User model
- Add validation for the Post model
- Add unit specs for all of your models' methods and validations.

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -12,6 +12,13 @@
 #  likes_count    :integer
 #
 class Post < ApplicationRecord
+  attribute :comments_count, :integer, default: 0
+  attribute :likes_count, :integer, default: 0
+
+  validates :title, presence: true, length: { in: 1..5 }
```
Comment (Deepakdanger) [inline app/models/post.rb]: - [ ] Validaton of title length should be 1 to 250 char.

**Review by Deepakdanger — CHANGES_REQUESTED** [review]
Hi @akos29, 

Good job so far! 👏
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight :100: 
- No linter error :heavy_check_mark: 
- Website looks professional :+1: 

## Required Changes ♻️
 - [ ] Kindly add some unit tests for the `User`, `Post`, `Comment`, and `Like` model
-  Any custom methods in the models should have a corresponding unit test.
- Kindly write all the validation test.
- [ ] _Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/post.rb [hunk 2]
```diff
@@ -12,6 +12,13 @@
...
+  attribute :comments_count, :integer, default: 0
+  attribute :likes_count, :integer, default: 0
+
+  validates :title, presence: true, length: { in: 1..250 }
+  validates :comments_count, comparison: { greater_than_or_equal_to: 0 }
+  validates :likes_count, comparison: { greater_than_or_equal_to: 0 }
+
   belongs_to :author, class_name: 'User', foreign_key: 'author_id', counter_cache: true
   has_many :comments
   has_many :likes
```
Comment (Kamba56) [inline app/models/post.rb:24]: - [ ] As requested by the previous reviewer, kindly add tests for all custom methods, this includes `show_recent_comments` and `update_posts_counter`.

File: app/models/user.rb [hunk 3]
```diff
@@ -11,6 +11,10 @@
 #  posts_count :integer
 #
 class User < ApplicationRecord
+  attribute :posts_count, :integer, default: 0
+  validates :name, presence: true, length: { minimum: 2 }
+  validates :posts_count, comparison: { greater_than_or_equal_to: 0 }
+
   has_many :posts, foreign_key: 'author_id'
   has_many :comments, foreign_key: 'author_id'
   has_many :likes, foreign_key: 'author_id'
```
Comment (Kamba56) [inline app/models/user.rb:20]: - [ ] As requested by the previous reviewer, kindly add tests for all custom methods, this includes `show_recent_posts`.

**Review by Kamba56 — CHANGES_REQUESTED** [review]
Hi @akos29, 

Great work on making the changes requested by a previous reviewer 👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
- [x] Great use of Github workflow!👏👏👏
- [x] All linters passed!👏👏👏
- [ ] Great job implementing unit tests for validations.

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by rotimiazeez — APPROVED** [review]
Hi @akos29 👏🏽, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlight
- Good job working on all suggested changes 👍🏽
- Tests are passing ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code. Some of them were simply missed by the previous reviewer, and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me(@rotimiazeez) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561), you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**